packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx

```diff
@@ -75,15 +75,16 @@ const PatientInfo: React.FC<PatientInfoProps> = ({ patient }) => {
   useEffect(() => {
     visitQueueEntries?.forEach((element) => {
       if (element?.patientUuid == patientUuid && currentVisit?.uuid === element.visitUuid) {
-        const visitQueueEntriesStatuses = {
-          Waiting: `${element.status} for ${element.service}`,
-          'In Service': `Attending ${element.service}`,
-          'Finished Service': `Finished ${element.service}`,
-        };
-        setCurrentService(visitQueueEntriesStatuses[element.status]);
         setVisitType(element.visitType);
         setPriority(element.priority);
         setQueueEntry(element);
+        if (element.status?.toLocaleLowerCase() === 'waiting') {
+          setCurrentService(`Waiting for ${element.service}`)
+        } else if (element.status?.toLocaleLowerCase() === 'in service') {
+          setCurrentService(`Attending ${element.service}`)
+        } else if (element.status?.toLocaleLowerCase() === 'finished service') {
+          setCurrentService(`Finished ${element.service}`)
+        }
```

Comment:
I would recommend to use `guard clause` here to avoid the nested `if ... else ` statement, which at times may be hard to maintain and read. You can check a reference [here](https://learningactors.com/javascript-guard-clauses-how-you-can-refactor-conditional-logic/)